src/CodeStyle/Core/Analyzers/Formatting/SourceTextExtensions.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Shared.Extensions
+{
+    internal static class SourceTextExtensions
```

Comment:
📝 Code was copied without modification from another file

Reply:
For the cases where things were copied without modification, can we use linking?

Reply:
Ok.  Here would be my strong preference.  Make these types partial.  Pull out the parts you want to share into a separate part.  Then link in that part.  The code duplication side of htis PR is pretty scary from a maintainability perspective.  IF we can at least use a linked file for the part of partials we want to share, that would help a lot.  we could even name the files in a useful manner.  Like SourceTextExtensions_CodeStyleParts.cs  or something...

Reply:
Tagging @jasonmalinowski for ideas/thoughts as well.

Reply:
➡️ Now shared